### PR TITLE
encodePNG implemented

### LINKED ISSUE #2: FSavePNG(io.Writer, image.Image) variant
I wanted to ask if adding an FSavePNG function would be something that you'd consider. The reason for this is, I can connect your package to a web server or for usage where I wouldn't like to write to disk and easily connect it to other usages since it will take an io.Writer.
Also this means that we can also reuse FSavePNG in SavePNG in such a fashion.
```go
func SavePNG(path string, img image.Image) error {
    f, err := os.Create(path)
    if err != nil {
      return err
    }
   defer f.Close()
   return FSavePNG(f, img)
}

func FSavePNG(w io.Writer, img image.Image) error {
    return png.Encode(w, img)
}
```

If this is something you'd consider, please let me know and I can create a PR for it.
Thank you.

### PR DESCRIPTION
**Pull request by odeke-em**
Fixes #2.

Allows passing in an io.Writer to save a PNG.

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -103,6 +104,12 @@ func (dc *Context) SavePNG(path string) error {
 	return SavePNG(path, dc.im)
 }
 
+// EncodePNG encodes the image as a PNG and
+// writes it to the provided io.Writer.
+func (dc *Context) EncodePNG(w io.Writer) error {
+	return EncodePNG(w, dc.im)
```
Comment (fogleman) [inline context.go]: This could just be `png.Encode(w, dc.im)` and then we don't even need the helper function in util.go.

File: context.go [hunk 2]
```diff
@@ -103,6 +104,12 @@ func (dc *Context) SavePNG(path string) error {
 	return SavePNG(path, dc.im)
 }
 
+// EncodePNG encodes the image as a PNG and
+// writes it to the provided io.Writer.
```
Comment (fogleman) [inline context.go]: Wrap this at 80 chars?

File: context.go (hunk 1 above)
Comment (odeke-em) [inline context.go]: Sounds like a plan. Coming up in a second.

Comment (odeke-em) [inline context.go]: Although just to confirm, do we want to encourage people to always use the Context methods? I was wondering if exposing the helper function would serve the same purpose as SavePNG does, ie usable by users of the package.

Comment (fogleman) [inline context.go]: True, but this one is so minimal I don't see much use for it.

File: context.go (hunk 2 above)
Comment (odeke-em) [inline context.go]: Done. Thanks.

File: context.go (hunk 1 above)
Comment (odeke-em) [inline context.go]: Gotcha. I've fixed those all up. PTAL.

File: context.go [hunk 3]
```diff
@@ -103,6 +106,14 @@ func (dc *Context) SavePNG(path string) error {
 	return SavePNG(path, dc.im)
 }
 
+// EncodePNG encodes the image as a PNG and writes it to the provided io.Writer.
+func (dc *Context) EncodePNG(w io.Writer) error {
+	if w == nil {
+		return fmt.Errorf("encodePNG: nil writer passed in")
```
Comment (fogleman) [inline context.go]: Why explicitly check for this? I assume `png.Encode` will fail.

Comment (odeke-em) [inline context.go]: png.Encode will crash :(

Comment (fogleman) [inline context.go]: I still think we can omit the check.

Comment (odeke-em) [inline context.go]: Aye aye, coming up in a second.

Comment (odeke-em) [inline context.go]: Updated, PTAL.

**Comment by fogleman** [comment]
There we go! Nice and simple.

**Comment by odeke-em** [comment]
Thank you @fogleman for the merge!

**Pull request merged**